builtin/providers/aws/resource_aws_elb.go

```diff
@@ -725,44 +725,61 @@ func resourceAwsElbUpdate(d *schema.ResourceData, meta interface{}) error {
 
 		removed := expandStringList(os.Difference(ns).List())
 		added := expandStringList(ns.Difference(os).List())
+		subnetconn := meta.(*AWSClient).ec2conn
 
-		if len(removed) > 0 {
-			detachOpts := &elb.DetachLoadBalancerFromSubnetsInput{
-				LoadBalancerName: aws.String(d.Id()),
-				Subnets:          removed,
+		// This section sorts the removed/added subnets based on availability zones
+		dup_az_count := 0
+		for i, r_subnet_id := range removed {
+			r_subnet, err := subnetconn.DescribeSubnets(&ec2.DescribeSubnetsInput{
+				SubnetIds: []*string{r_subnet_id},
+			})
+			if err != nil {
+				return err
+			}
+			for j, a_subnet_id := range added {
+				a_subnet, err := subnetconn.DescribeSubnets(&ec2.DescribeSubnetsInput{
+					SubnetIds: []*string{a_subnet_id},
+				})
+				if err != nil {
+					return err
+				}
+				if *r_subnet.Subnets[0].AvailabilityZone == *a_subnet.Subnets[0].AvailabilityZone {
+					removed[dup_az_count], removed[i] = removed[i], removed[dup_az_count]
+					added[j], added[dup_az_count] = added[dup_az_count], added[j]
+					dup_az_count += 1
+				}
 			}
+		}
 
-			log.Printf("[DEBUG] ELB detach subnets opts: %s", detachOpts)
-			_, err := elbconn.DetachLoadBalancerFromSubnets(detachOpts)
+		// Subnets with the same availability zones need to be removed
+		// and then added one at a time to avoid api errors
+		for i := 0; i < dup_az_count; i++ {
+			curr_removed_subnet := make([]*string, 0, 1)
+			curr_removed_subnet = append(curr_removed_subnet, removed[i])
```

Comment:
Nit: this could just be `curr_removed_subnet := []*string{removed[i]}`, unless I'm missing something.